openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -0,0 +1,132 @@
+from django.contrib.auth.models import User
+from rest_framework.response import Response
+
+from .constants import GROUP_TRAINING_MANAGERS
+from .serializers import UserSerializer
+from rest_framework import viewsets, status
+
+from rest_framework.decorators import permission_classes
+
+from .permissions import CanAccessPakXAdminPanel
+
+
+@permission_classes([CanAccessPakXAdminPanel])
+class UserProfileViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+    http_method_names = ['get', 'post', 'delete']
+
+    def create(self, request, *args, **kwargs):
+
+        try:
+            activation_status = self.get_activation_status(request.data["action"])
+            if activation_status is not None:
+                if request.data["ids"] == "all":
+
+                    all_users = User.objects.all()
+
+                    for user in all_users:
+                        User.objects.filter(id=user.id).update(is_active=activation_status)
+
+                    return Response(status=status.HTTP_200_OK)
+
+                else:
+                    for update_id in request.data["ids"]:
+                        if not User.objects.filter(id=update_id).update(is_active=activation_status):
+                            return Response({"ID not found": update_id}, status=status.HTTP_404_NOT_FOUND)
+
+                    return Response(status=status.HTTP_200_OK)
+            else:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+        except:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+    def destroy(self, request, *args, **kwargs):
+        if User.objects.filter(id=self.get_object().id).update(is_active=False):
+            return Response(status=status.HTTP_200_OK)
+        else:
+            return Response({"ID not found": self.get_object().id}, status=status.HTTP_404_NOT_FOUND)
+
+    def list(self, request, *args, **kwargs):
+        if len(self.request.query_params) == 0:
+            return Response(self.serializer_class(self.queryset.order_by('-id'), many=True).data)
+
+        else:
+            queryset = None
+
+            if self.request.query_params.get('roles', None) is not None:
+                roles = self.request.query_params.get('roles').split(',')
+                queryset = self.filter_users_by_roles(roles, queryset)
```

Comment:
queryset shouldn't be passed in functions, we already have a queryset at class level.